### PR TITLE
fixes #14389 - import/export host groups

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

File: lib/hammer_cli_csv/host_groups.rb [hunk 1]
```diff
@@ -45,66 +47,77 @@ def export
             realm = hostgroup['realm_name']
             puppet_proxy = hostgroup['puppet_proxy_id'] ? foreman_host(:id => hostgroup['puppet_proxy_id']) : nil
             puppet_ca_proxy = hostgroup['puppet_ca_proxy_id'] ? foreman_host(:id => hostgroup['puppet_ca_proxy_id']) : nil
-            # TODO: http://projects.theforeman.org/issues/7597
-            # content_source = hostgroup['content_source_id'] ? foreman_host(:id => hostgroup['content_source_id']) : nil
-            content_source = nil
+            content_source = hostgroup['content_source_id'] ? foreman_host(:id => hostgroup['content_source_id']) : nil
```
Comment (beav) [inline lib/hammer_cli_csv/host_groups.rb]: the TODO was removed, but it looks like the ticket itself is still in NEW state. Is the TODO no longer relevant?

File: lib/hammer_cli_csv/host_groups.rb [hunk 2]
```diff
@@ -45,66 +47,77 @@ def export
...
+      def create_from_csv(line)
         return if option_organization && !CSV.parse_line(line[ORGANIZATIONS], {:skip_blanks => true}).include?(option_organization)
 
+        params = {
+          'hostgroup' => {
+            'architecture_id' => foreman_architecture(:name => line[ARCHITECTURE]),
+            'operatingsystem_id' => foreman_operatingsystem(:name => line[OPERATING_SYSTEM]),
+            'medium_id' => foreman_medium(:name => line[MEDIUM]),
+            'ptable_id' => foreman_partitiontable(:name => line[PARTITION_TABLE]),
+            'medium_id' => foreman_medium(:name => line[MEDIUM]),
```
Comment (beav) [inline lib/hammer_cli_csv/host_groups.rb]: `medium_id` is in here 2x

File: lib/hammer_cli_csv/host_groups.rb (hunk 1 above)
Comment (thomasmckay) [inline lib/hammer_cli_csv/host_groups.rb]: Good catch, thanks! I added a needinfo to the redmine. The issue no longer exists for my usage of the API here so the TODO is no longer relevant.

File: lib/hammer_cli_csv/host_groups.rb (hunk 2 above)
Comment (thomasmckay) [inline lib/hammer_cli_csv/host_groups.rb]: corrected

File: lib/hammer_cli_csv/host_groups.rb [hunk 3]
```diff
@@ -45,66 +47,76 @@ def export
...
           end
+
+          # TODO: puppet classes
+          puppetclass_ids = collect_column(line[PUPPET_CLASSES]) do |puppet_class|
+            module_name, name = puppet_class.split('/')
+            foreman_puppet_class(:name => name)
+          end
+          existing_ids = hostgroup['puppet_classes'].collect { |puppet_class| puppet_class['id'] }
+          # DELETE existing_ids - puppetclass_ids
+          # POST puppetclass_ids - existing_ids
```
Comment (beav) [inline lib/hammer_cli_csv/host_groups.rb]: I am not sure what these comments are for, are they supposed to be in the PR?

**Comment by beav** [comment]
@thomasmckay i made a small comment but looks good overall

File: lib/hammer_cli_csv/host_groups.rb (hunk 3 above)
Comment (thomasmckay) [inline lib/hammer_cli_csv/host_groups.rb]: Export/import host groups is pretty complicated and will remain in the "unsupported" category for a while. The TODO comment above will remain unfixed in this PR and the two comments are what needs to happen. I think. Not all the foreman APIs accept adding relationships versus fully resetting them.

tl;dr - more work to do in future

**Comment by beav** [comment]
ACK
